core/src/lightclient/mod.rs

```diff
@@ -354,4 +367,131 @@ impl StarknetClient for StarknetClientImpl {
         let eth_tx = starknet_tx_into_eth_tx(starknet_tx)?;
         Ok(eth_tx)
     }
+
+    /// Returns the receipt of a transaction by transaction hash.
+    ///
+    /// # Arguments
+    ///
+    /// * `hash(H256)` - The block hash.
+    ///
+    /// # Returns
+    ///
+    ///  * `transaction_receipt(TransactionReceipt)` - The transaction receipt.
+    ///
+    /// `Ok(Option<TransactionReceipt>)` if the operation was successful.
+    /// `Err(LightClientError)` if the operation failed.
+    async fn get_transaction_receipt(
+        &self,
+        hash: H256,
+    ) -> Result<Option<TransactionReceipt>, LightClientError> {
+        let mut res_receipt = create_default_transaction_receipt();
+
+        //TODO: Error when trying to transform 32 bytes hash to FieldElement
+        let hash_hex = hex::encode(hash);
+        let hash_felt = FieldElement::from_hex_be(&hash_hex).map_err(|e| {
+            LightClientError::OtherError(anyhow::anyhow!(
+                "Failed to convert Starknet block hash to FieldElement: {}",
+                e
+            ))
+        })?;
+        let starknet_tx_receipt = self.client.get_transaction_receipt(hash_felt).await?;
+
+        match starknet_tx_receipt {
+            MaybePendingTransactionReceipt::Receipt(receipt) => {
+                match receipt {
+                    StarknetTransactionReceipt::Invoke(InvokeTransactionReceipt {
+                        transaction_hash,
+                        actual_fee,
+                        status,
+                        block_hash,
+                        block_number,
+                        ..
+                    })
+                    | StarknetTransactionReceipt::L1Handler(L1HandlerTransactionReceipt {
```

Comment:
I think you can refactor this `match` to work without `|` but rather with different matching arms.

We should throw an error for now on L1Handler, DeclareTransaction, as we don't support them

Reply:
Good for me!